les/handler.go

```diff
@@ -429,29 +431,34 @@ func (pm *ProtocolManager) handleMsg(p *peer) error {
 			// Retrieve the next header satisfying the query
 			var origin *types.Header
 			if hashMode {
-				origin = pm.blockchain.GetHeaderByHash(query.Origin.Hash)
+				if first {
```

Comment:
exactly the same code as here: https://github.com/ethereum/go-ethereum/pull/16946/files#diff-74cea5b79d38ad655dd2c0b1279bc7e7R356
perhaps extract to a function? very WET otherwise

Reply:
Makes sense but I wanted to do the least amount of change in this hotfix PR.

Reply:
ACK - perhaps create a follow-up refactoring issue then to not forget about it ?-)